mantle/platform/metal.go

```diff
@@ -713,36 +700,22 @@ func (inst *Install) InstallViaISOEmbed(kargs []string, liveIgnition, targetIgni
 			return nil, errors.Wrapf(err, "running coreos-installer iso kargs modify")
 		}
 		srcisopath = newIso
-		copyNetworkArg = "--copy-network"
+		installerConfig.CopyNetwork = true
 	}
 
-	insecureOpt := ""
 	if inst.Insecure {
-		insecureOpt = "--insecure"
+		installerConfig.Insecure = true
 	}
-	pointerIgnitionPath := "/var/opt/pointer.ign"
 
-	installerUnit := fmt.Sprintf(`
-[Unit]
-Description=TestISO CoreOS Installer
-After=network-online.target
-Wants=network-online.target
-OnFailure=emergency.target
-OnFailureJobMode=isolate
-[Service]
-RemainAfterExit=yes
-Type=oneshot
-ExecStart=/usr/bin/coreos-installer install %s --ignition %s %s %s %s %s
-[Install]
-WantedBy=multi-user.target
-`, srcOpt, pointerIgnitionPath, insecureOpt, targetDevice, appendMultipathKargs, copyNetworkArg)
+	installerConfigData, err := yaml.Marshal(installerConfig)
+	if err != nil {
+		return nil, err
+	}
 	mode := 0644
-	rebootUnitP := string(rebootUnit)
 
-	inst.liveIgnition.AddSystemdUnit("coreos-installer.service", installerUnit, conf.Enable)
-	inst.liveIgnition.AddSystemdUnit("coreos-installer-reboot.service", rebootUnitP, conf.Enable)
 	inst.liveIgnition.AddSystemdUnit("boot-started.service", bootStartedUnit, conf.Enable)
-	inst.liveIgnition.AddFile(pointerIgnitionPath, serializedTargetConfig, mode)
+	inst.liveIgnition.AddFile(installerConfig.IgnitionFile, serializedTargetConfig, mode)
+	inst.liveIgnition.AddFile("/etc/coreos/installer.d/mantle", string(installerConfigData), mode)
```

Comment:
Minor/optional: small preference for

```suggestion
	inst.liveIgnition.AddFile("/etc/coreos/installer.d/mantle.yaml", string(installerConfigData), mode)
```

:smiley: